TCAT/Controllers/HomeOptionsCardViewController+Extensions.swift

```diff
@@ -0,0 +1,353 @@
+            case .recentSearches:
+                let place = sections[indexPath.section].getItems()[indexPath.row]
+                recentLocations = Global.shared.deleteRecent(recent: place, allRecents: recentLocations)
+                createDefaultSections()
+            default: break
+            }
+
+        }
+    }
+
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+
+        let routeOptionsViewController = RouteOptionsViewController()
+        routeOptionsViewController.didReceiveCurrentLocation(currentLocation)
+        let allStopsTableViewConroller = AllStopsTableViewController()
+        var didSelectAllStops = false
+        var shouldPushViewController = true
+
+        if sections[indexPath.section] == .seeAllStops {
+            didSelectAllStops = true
+        } else {
+            let place = sections[indexPath.section].getItems()[indexPath.row]
+            if place.name == Constants.General.firstFavorite {
+                shouldPushViewController = false
+                presentFavoritesTVC()
+            } else {
+                routeOptionsViewController.searchTo = place
+                Global.shared.insertPlace(for: Constants.UserDefaults.recentSearch, place: place)
+            }
+        }
+
+        tableView.deselectRow(at: indexPath, animated: true)
+        searchBar.endEditing(true)
+
+        let vcToPush = didSelectAllStops ? allStopsTableViewConroller : routeOptionsViewController
+        if shouldPushViewController {
+            navigationController?.pushViewController(vcToPush, animated: true)
+        }
+    }
+}
+
+// MARK: DZN Empty Data Set Source
+extension HomeOptionsCardViewController: DZNEmptyDataSetSource, DZNEmptyDataSetDelegate {
+    func verticalOffset(forEmptyDataSet scrollView: UIScrollView) -> CGFloat {
+        // If tableview header is hidden, increase offset to center EmptyDataSet view
+        return tableView.tableHeaderView == nil ? -80 : (-80 - tableView.contentInset.top)
+    }
+
+    func image(forEmptyDataSet scrollView: UIScrollView) -> UIImage? {
+        // If loading indicator is being shown, don't display image
+        if isLoading { return nil }
+        return isNetworkDown ? #imageLiteral(resourceName: "noWifi") : #imageLiteral(resourceName: "noRoutes")
+    }
+
+    func description(forEmptyDataSet scrollView: UIScrollView) -> NSAttributedString? {
+        // If loading indicator is being shown, don't display description
+        if isLoading { return nil }
+        let title = isNetworkDown ? Constants.EmptyStateMessages.noNetworkConnection: Constants.EmptyStateMessages.locationNotFound
+        return NSAttributedString(string: title, attributes: [.foregroundColor: Colors.metadataIcon])
+    }
+
+    func buttonTitle(forEmptyDataSet scrollView: UIScrollView, for state: UIControl.State) -> NSAttributedString? {
+        // If loading indicator is being shown, don't display button
+        if isLoading { return nil }
```

Comment:
Is there no way to have all of these `isLoading` calls in one place?

Reply:
Will remove it completely. I noticed we were never even setting the delegates to self and thus were never even using all of the loading indicator stuff.